clients/src/main/java/org/apache/kafka/common/resource/PatternType.java

```diff
@@ -26,51 +26,56 @@
 import java.util.stream.Collectors;
 
 /**
- * Resource name type.
+ * Resource pattern type.
  */
 @InterfaceStability.Evolving
-public enum ResourceNameType {
+public enum PatternType {
     /**
-     * Represents any ResourceNameType which this client cannot understand, perhaps because this client is too old.
+     * Represents any PatternType which this client cannot understand, perhaps because this client is too old.
      */
     UNKNOWN((byte) 0),
 
     /**
-     * In a filter, matches any resource name type.
+     * In a filter, matches any resource pattern type.
      */
     ANY((byte) 1),
```

Comment:
We should rename this to "MATCH" as per our earlier discussion, since its behavior doesn't match the other "ANY" fields

Reply:
MATCH is the next one down.